src/App.tsx

```diff
@@ -1,24 +1,212 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
 
-const USER_ID = 0;
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { getVisibleTodos } from './utils/helpers';
+
+const USER_ID = 10930;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoStatus>(TodoStatus.ALL);
+  const [error, setError] = useState<string | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch((fetchedError: Error) => {
+        setError(fetchedError?.message ?? 'Unexpected error fetching todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeout: number;
+
+    if (error) {
+      timeout = window.setTimeout(() => setError(null), 3000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [error]);
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return getVisibleTodos(filter, todos, completedTodos, activeTodos);
+  }, [todos, filter]);
+
+  const handleAddTodo = useCallback(async (todoTitle: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
```

Comment:
Adding a `catch` block without catching the error is a bad practice, because now we don't give the user an opportunity to understand what's wrong and only give a general answer ( `catch (error)` ) .
This applies to all code.